vktrace/src/vktrace_common/optimization_function.cpp

```diff
@@ -0,0 +1,55 @@
+#include <stdio.h>
+#include <string.h>
+#include <ppl.h>
+#include <stdlib.h>
+
+using namespace concurrency;
+using namespace std;
+
+
+#define PARALLEL_INVOKE_NUM   10
+#define SIZE_LIMIT_TO_USE_OPTIMIZATION 67100000
+extern "C" void *opt_memcpy(void * destination, const void * source, size_t size)

```

Comment:
Parallel_Invoke should handle the threading side for us based on the specific system; in this case it will use up to 10 threads if they are available otherwise it will partition the work among the necessary threads.